### ISSUE TITLE
Validation Failed

### ISSUE THREAD

**Issue by JacobBangSSE** [scandihealth/lpr3-docs#49]
I get this error:
```xml
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
   <env:Header xmlns:env="http://www.w3.org/2003/05/soap-envelope"/>
   <soap:Body>
      <soap:Fault>
         <soap:Code>
            <soap:Value>soap:Receiver</soap:Value>
         </soap:Code>
         <soap:Reason>
            <soap:Text xml:lang="en">Validation Failed.</soap:Text>
         </soap:Reason>
      </soap:Fault>
   </soap:Body>
</soap:Envelope>
```

When I try send the following to PatientHealthcareValidateReportingService:
[http_post_request3.txt](https://github.com/scandihealth/lpr3-docs/files/1758835/http_post_request3.txt)

Is it possible to get more information about what part of the Envelope which did not validate?

**Comment by MartinHylleberg** [comment]
Any updates on this one?

**Comment by TueCN** [comment]
Hi

It looks like the IDCard could not be validated. The signature is reported to be invalid by Seal.Java 2.3.1.
We will look into returning a SOAP Fault that better describes this problem.

I think you are running into the issue from #44. I can see you have raised issue #48 to discuss how this should be handled.

In the mean time, you could try and only apply MTOM to the document itself (i.e. not the signature), if you want to further test the service.

**Comment by JacobBangSSE** [comment]
If you read the attached post request you can see I don't use MTOM on the certificate but only on the document itself. It is correct we have made issue #44 and #48 and they are still relevant, but to make some progress in our tests we have manually made some changes to the messages so the certificate are are no longer attached as MTOM (the fix cannot be used in production but should still be usable in tests).

So I want to ask again if we can get more details about why we get a "Validation Failed." error.

**Comment by TueCN** [comment]
Ok, so it is not an MTOM problem.

The only thing we can say is that your signature is invalid.

We have tried to validate your attached message against http://test2.ekstern-test.nspop.dk:8080/nts/service and get the following response:

```xml
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
   <SOAP-ENV:Header>
      <Header xmlns="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd">
         <Linking>
            <FlowID>6956</FlowID>
            <MessageID>6956:9929771679131155</MessageID>
            <InResponseToMessageID>d25c08efc0a8380100a6a684d6bbd32d</InResponseToMessageID>
         </Linking>
         <FlowStatus>syntax_error</FlowStatus>
      </Header>
      <NTS xmlns="http://nspop.dk/nts/2013/05">
         <Environment>TEST</Environment>
         <NTS-Version>1.0.2</NTS-Version>
         <SEAL-Version>2.1.6</SEAL-Version>
      </NTS>
   </SOAP-ENV:Header>
   <SOAP-ENV:Body>
      <SOAP-ENV:Fault>
         <faultcode>Server</faultcode>
         <faultstring>Consult detailed errors in medcom:FaultCode</faultstring>
         <detail xmlns:medcom="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd">
            <medcom:FaultCode>ERROR [DGWS IDCard]: IDCard Conditions NotOnOrAfter exceeded (&lt;saml:Conditions xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" NotBefore="2018-02-26T13:40:01Z" NotOnOrAfter="2018-02-27T13:40:01Z"/>). Server time: 2018-03-01T14:07:04.036</medcom:FaultCode>
            <medcom:FaultCode>ERROR [DGWS IDCard]: Inconsistent: IDCard AuthenticationLevel set to 3, DGWS SecurityLevel set to 0</medcom:FaultCode>
            <medcom:FaultCode>ERROR [STS Signed]: SEAL (Test Federation): Invalid signature: 'dk.sosi.seal.modelbuilders.SignatureInvalidModelBuildException: Signature could not be validated'</medcom:FaultCode>
            <medcom:FaultCode>ERROR [STS Signed]: IDCard contains more than 4 line breaks. This indicates that the IDCard XML has been pretty printed</medcom:FaultCode>
         </detail>
      </SOAP-ENV:Fault>
   </SOAP-ENV:Body>
</SOAP-ENV:Envelope>
```

The interesting error here is: **ERROR [STS Signed]: SEAL (Test Federation): Invalid signature: 'dk.sosi.seal.modelbuilders.SignatureInvalidModelBuildException: Signature could not be validated'**

When we call the same service with a message we have signed ourself we get:
```xml
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">
   <SOAP-ENV:Header>
      <Header xmlns="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd">
         <Linking>
            <FlowID>389A</FlowID>
            <MessageID>389A:9930171693376344</MessageID>
            <InResponseToMessageID>AAABYbihX63aq8iLuo78a1NPU0k=</InResponseToMessageID>
         </Linking>
         <FlowStatus>flow_finalized_succesfully</FlowStatus>
      </Header>
      <NTS xmlns="http://nspop.dk/nts/2013/05">
         <Environment>TEST</Environment>
         <NTS-Version>1.0.2</NTS-Version>
         <SEAL-Version>2.1.6</SEAL-Version>
      </NTS>
   </SOAP-ENV:Header>
   <SOAP-ENV:Body>
      <SOAP-ENV:Fault>
         <faultcode>Server</faultcode>
         <faultstring>Consult detailed errors in medcom:FaultCode</faultstring>
         <detail xmlns:medcom="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd">
            <medcom:FaultCode>ERROR [DGWS IDCard]: IDCard Conditions NotOnOrAfter exceeded (&lt;saml:Conditions xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" NotBefore="2018-02-21T13:45:36Z" NotOnOrAfter="2018-02-22T13:45:36Z"/>). Server time: 2018-03-01T14:13:44.016</medcom:FaultCode>
         </detail>
      </SOAP-ENV:Fault>
   </SOAP-ENV:Body>
</SOAP-ENV:Envelope>
```
Which is just an error to say our IDCard has expired (but signature is not invalid)

**Comment by JacobBangSSE** [comment]
Thanks for you suggestion about testing against the nspop service to resolve the STS certificate problems we have. I think we found the problem which turned out to be we missed to specify the "DGWS SecurityLevel". The error about the 4 line breaks is because I have pretty printet the XML to make it more readable for you. After I fixed this two issues I got past the nspop validation service.

Now we gets a new error message but I am going to create a new issue about that. :)
